### ISSUE TITLE
[PhotoSync] Track upload/retry status of individual photos

### ISSUE THREAD

**Issue by scolsen** [google/ground-android#2120]
Right now, we batch process and batch retry photo uploads. This means we attempt to re-upload photos when one upload in the batch fails even if the other photos in question were successfully uploaded. Tracking upload status on a per-photo basis instead will allow us to avoid this duplicative work.

**Comment by scolsen** [comment]
@rawbzz -- could use some UX help in figuring out the best way to indicate the status of individual photo uploads when you get a chance! Each photo upload can succeed/fail individually, and each submission may have one or more photo associated with it. lmk if I can provide more context, thanks!

**Comment by rawbzz** [comment]
@scolsen Yes! Do you have any screenshots of what currently happens right now?

**Comment by gino-m** [comment]
> @scolsen Yes! Do you have any screenshots of what currently happens right now?

This is lower priority, so removing milestone and bumping to P2. This is the redesigned status screen which we're targeting for next week:

![image](https://github.com/google/ground-android/assets/228050/6d7c8361-e547-4964-aaab-66c482e416e8)

**Comment by gino-m** [comment]
This code:

https://github.com/google/ground-android/blob/17422bdd7f9b3cd1b8bdec94f0a6f1f4a10a5245/ground/src/main/java/com/google/android/ground/persistence/sync/MediaUploadWorker.kt#L117-L122

Is incorrect in that the calling function decided the retry behavior based on only one of the photo upload results. Reclassifying as bug.

**Comment by gino-m** [comment]
@scolsen Adding as low priority for Iteration 27.
